software/messaging/src/main/java/brooklyn/entity/messaging/rabbit/RabbitSshDriver.java

```diff
@@ -73,7 +81,15 @@ public void install() {
         List<String> urls = resolver.getTargets();
         String saveAs = resolver.getFilename();
 
+        String osMajorVersion = getMachine().getOsDetails().getVersion();
+        osMajorVersion = osMajorVersion.indexOf(".") > 0 ? osMajorVersion.substring(0, osMajorVersion.indexOf('.')) : osMajorVersion;
```

Comment:
`osMajorVersion` and `osArchitecture` are both annotated `@Nullable` so there's a chance of `NullPointerExceptions` here.